clients/rest-client-generator/src/main/java/uk/gov/justice/services/clients/rest/generator/strategy/ClientActionMappingGeneration.java

```diff
@@ -0,0 +1,51 @@
+package uk.gov.justice.services.clients.rest.generator.strategy;
+
+import uk.gov.justice.raml.common.mapper.ActionMapping;
+
+import java.util.Collection;
+import java.util.List;
+
+import org.raml.model.ActionType;
+import org.raml.model.MimeType;
+
+/**
+ * Action mapping generation strategy to support Action Mapping.
+ */
+public class ClientActionMappingGeneration implements ClientGenerationStrategy {
+
+    /**
+     * The list of {@link ActionMapping} parsed from the description of a RAML Action.
+     *
+     * @param description the description from a RAML Action
+     * @return a list of {@link ActionMapping}
+     */
+    @Override
+    public List<ActionMapping> listOfActionMappings(final String description) {
+        return ActionMapping.listOf(description);
+    }
+
+    /**
+     * Returns the {@link ActionMapping} from the list for the given MimeType and ActionType.
+     *
+     * @param actionMappings the list of {@link ActionMapping}
+     * @param mimeType       the MimeType to use for the lookup
+     * @param httpMethod     the ActionType to use for the lookup
+     * @return the {@link ActionMapping}
+     */
+    @Override
+    public ActionMapping mappingOf(final Collection<ActionMapping> actionMappings, final MimeType mimeType, final ActionType httpMethod) {
+        return actionMappings.stream().filter(m -> m.mimeTypeFor(httpMethod).equals(mimeType.getType())).findAny().get();
+    }
+
+    /**
+     * Returns the handles annotation value taken from the {@link ActionMapping} name.
+     *
+     * @param mapping the {@link ActionMapping} to use.
+     * @param header  the header is not used for this strategy.
+     * @return the handles annotation value
+     */
+    @Override
+    public String handlesValue(final ActionMapping mapping, final String header) {
```

Comment:
I have had to use a common interface to achieve the strategy switching mechanism for backward compatibility.  This code should be temporary and removed once all contexts are converted to use action mapping.